GoogleChromeLabs/ps-analysis-tool
# Enhancement: Implement multi auction scenario

## Description
This PR updates the processing logic for PA data and the UI elements to render the data.
It adds a sidebar to the Auction table, allowing users to navigate multiple ad units and multiple auctions for single/multi-sellers.


## Relevant Technical Choices
- A new type has been introduced named `AuctionEventsType`
```
type AuctionEventsType = {
  [adunit: string]: {
    [time: string]: {
      [sellerURL: string]: {
        [auctionHostURL: string]: singleAuctionEvent[];
      };
    };
  };
};
```


## Testing Instructions
- Open the PSAT panel and navigate to Protected Audience -> Auctions panel.
- Navigate to https://privacysandboxdemos.domain-aaa.com/, join IG, and run multiple auctions.
- The auction panel should update and show multiple auctions.
- Now try it in a multi-seller scenario.


## Additional Information:



## Screenshot/Screencast

https://github.com/user-attachments/assets/fea92b16-c47a-4c32-b3fc-759ee7e895ea






---

## Checklist






Fixes #786
